Use C++20 std::ranges::sort in JavaScriptCore

iidmsa opened the pull request:
#### 38a918f22e55503276704745b0cc0b1278d96b42
<pre>
Use C++20 std::ranges::sort in JavaScriptCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=302761">https://bugs.webkit.org/show_bug.cgi?id=302761</a>
<a href="https://rdar.apple.com/165017405">rdar://165017405</a>

Reviewed by Sam Weinig.

* Source/JavaScriptCore/assembler/ARM64Assembler.h:
* Source/JavaScriptCore/assembler/ARMv7Assembler.h:
(JSC::ARMv7Assembler::jumpsToLink):
* Source/JavaScriptCore/b3/B3InferSwitches.cpp:
* Source/JavaScriptCore/b3/B3LowerMacros.cpp:
* Source/JavaScriptCore/b3/B3OptimizeAssociativeExpressionTrees.cpp:
(JSC::B3::OptimizeAssociativeExpressionTrees::optimizeRootedTree):
* Source/JavaScriptCore/b3/B3UseCounts.cpp:
(JSC::B3::UseCounts::UseCounts):
* Source/JavaScriptCore/b3/B3Validate.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackByLinearScan.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersByGreedy.cpp:
(JSC::B3::Air::Greedy::GreedyAllocator::finalizeGroups):
* Source/JavaScriptCore/b3/air/AirAllocateStackByGraphColoring.cpp:
* Source/JavaScriptCore/b3/air/AirFixObviousSpills.cpp:
* Source/JavaScriptCore/bytecode/CallLinkStatus.cpp:
(JSC::CallLinkStatus::computeFromCallLinkInfo):
* Source/JavaScriptCore/bytecode/PreciseJumpTargets.cpp:
(JSC::computePreciseJumpTargetsInternal):
* Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp:
(JSC::MetadataStatistics::reportMetadataStatistics):
* Source/JavaScriptCore/debugger/DebuggerParseData.cpp:
(JSC::DebuggerPausePositions::forEachBreakpointLocation):
(JSC::DebuggerPausePositions::sort):
* Source/JavaScriptCore/dfg/DFGCommonData.cpp:
(JSC::DFG::CommonData::finalizeCatchEntrypoints):
* Source/JavaScriptCore/dfg/DFGGraph.cpp:
(JSC::DFG::Graph::dumpBlockHeader):
* Source/JavaScriptCore/dfg/DFGJITCode.cpp:
(JSC::DFG::JITCode::finalizeOSREntrypoints):
* Source/JavaScriptCore/dfg/DFGLoopUnrollingPhase.cpp:
(JSC::DFG::LoopUnrollingPhase::populateCandidateLoops):
* Source/JavaScriptCore/dfg/DFGMinifiedGraph.cpp:
(JSC::DFG::MinifiedGraph::prepareAndShrink):
* Source/JavaScriptCore/dfg/DFGNode.h:
(JSC::DFG::nodeMapDump):
(JSC::DFG::nodeValuePairListDump):
* Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp:
* Source/JavaScriptCore/dfg/DFGScoreBoard.h:
(JSC::DFG::ScoreBoard::sortFree):
* Source/JavaScriptCore/dfg/DFGStoreBarrierClusteringPhase.cpp:
* Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp:
(JSC::FTL::DFG::LowerDFGToB3::compileCompareStrictEq):
* Source/JavaScriptCore/heap/HeapSnapshot.cpp:
(JSC::HeapSnapshot::finalize):
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
(JSC::HeapSnapshotBuilder::dumpToStream):
* Source/JavaScriptCore/heap/JITStubRoutineSet.cpp:
(JSC::JITStubRoutineSet::prepareForConservativeScan):
* Source/JavaScriptCore/heap/MarkedSpace.cpp:
* Source/JavaScriptCore/heap/MarkingConstraintSet.cpp:
(JSC::MarkingConstraintSet::executeConvergenceImpl):
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
(Inspector::JSInjectedScriptHost::queryHolders):
* Source/JavaScriptCore/jit/JITSizeStatistics.cpp:
(JSC::JITSizeStatistics::dump const):
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
(JSC::CompactTDZEnvironment::sortCompact):
* Source/JavaScriptCore/runtime/ArrayPrototype.cpp:
(JSC::sortBucketSort):
* Source/JavaScriptCore/runtime/ImportMap.cpp:
(JSC::ImportMap::ImportMap):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::flattenFields):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::availableCollations):
(JSC::availableCurrencies):
(JSC::availableNumberingSystems):
(JSC::intlAvailableTimeZones):
* Source/JavaScriptCore/runtime/JSModuleNamespaceObject.cpp:
(JSC::JSModuleNamespaceObject::finishCreation):
* Source/JavaScriptCore/runtime/JSONObject.cpp:
(JSC::logOutcomeImpl):
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::JSObject::getOwnIndexedPropertyNames):
* Source/JavaScriptCore/runtime/JSObjectInlines.h:
(JSC::JSObject::forEachOwnIndexedProperty):
* Source/JavaScriptCore/wasm/WasmMergedProfile.cpp:
(JSC::Wasm::MergedProfile::Candidates::finalize const):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::prepareForTailCallImpl):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::prepareForTailCallImpl):
* Source/JavaScriptCore/wasm/WasmOpcodeCounter.cpp:
(JSC::Wasm::WasmOpcodeCounter::dump):
* Source/JavaScriptCore/wasm/debugger/tests/TestUtilities.cpp:
(WasmDebugInfoTest::parseAndVerifyDebugInfoImpl):
* Source/JavaScriptCore/yarr/YarrJIT.cpp:
* Source/JavaScriptCore/yarr/YarrPattern.cpp:
(JSC::Yarr::CharacterClassConstructor::performSetOpWithMatches):
(JSC::Yarr::CharacterClassConstructor::sort):

Canonical link: <a href="https://commits.webkit.org/303399@main">https://commits.webkit.org/303399@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4c32d9ab165f4cbdedfd71e6dfcabfc58730842

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132219 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4712 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43247 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139734 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84146 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4664 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4473 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101063 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68381 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3295 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118434 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81858 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3187 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82954 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/124282 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111989 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36553 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142381 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/130726 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4381 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37133 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109441 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4462 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3841 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109622 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27787 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3318 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114706 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57628 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4435 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33082 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163693 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4267 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67881 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/42535 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4526 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4394 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/JavaScriptCore/b3/B3LowerMacros.cpp:
```diff
@@ -287,11 +287,9 @@ class LowerMacros {
-                    [] (const SwitchCase& left, const SwitchCase& right) {
-                        return left.caseValue() < right.caseValue();
-                    });
+                std::ranges::sort(cases, [] (const SwitchCase& left, const SwitchCase& right) {
```
Please remove the space between the `[]` and the parameter list. Also can probably use auto here.

```suggestion
                std::ranges::sort(cases, [](auto& left, auto& right) {
```

weinig commented on Source/JavaScriptCore/assembler/ARM64Assembler.h:
```diff
@@ -3869,7 +3869,7 @@ class ARM64Assembler {
     Vector<LinkRecord, 0, UnsafeVectorOverflow>& jumpsToLink()
     {
-        std::sort(m_jumpsToLink.begin(), m_jumpsToLink.end(), [](auto& a, auto& b) {
+        std::ranges::sort(m_jumpsToLink, [](auto& a, auto& b) {
```
It's not necessary for this change to go through, but something to consider is using the projection parameter to simplify these a bit. For instance, I believe this could be written as:

```c++
        std::ranges::sort(m_jumpsToLink, {}, &LinkRecord::from);
```

weinig commented on Source/JavaScriptCore/b3/B3OptimizeAssociativeExpressionTrees.cpp:
```diff
@@ -191,7 +191,7 @@ bool OptimizeAssociativeExpressionTrees::optimizeRootedTree(Value* root, Inserti
     }
 
-    std::sort(leaves.begin(), leaves.end(), [](Value* x, Value* y) {
+    std::ranges::sort(leaves, [](Value* x, Value* y) {
```
Can update this to use auto instead.

```suggestion
    std::ranges::sort(leaves, [](auto* x, auto* y) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackByLinearScan.cpp:
```diff
@@ -300,11 +300,9 @@ class LinearScan {
-            [] (Clobber& a, Clobber& b) -> bool {
-                return a.index < b.index;
-            });
+        std::ranges::sort(m_clobbers, [] (Clobber& a, Clobber& b) -> bool {
```
Can remove the space, use auto, and remove the -> bool.

```suggestion
        std::ranges::sort(m_clobbers, [](auto& a, auto& b) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackByLinearScan.cpp:
```diff
@@ -369,11 +367,9 @@ class LinearScan {
-            [&] (Tmp& a, Tmp& b) {
-                return m_map[a].interval.begin() < m_map[b].interval.begin();
-            });
+        std::ranges::sort(m_tmps, [&] (Tmp& a, Tmp& b) {
```
```suggestion
        std::ranges::sort(m_tmps, [&](auto& a, auto& b) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirAllocateRegistersByGreedy.cpp:
```diff
@@ -1164,8 +1164,7 @@ class GreedyAllocator {
             TmpData& data = m_map[tmp];
-            std::sort(data.coalescables.begin(), data.coalescables.end(),
-                [this] (const auto& a, const auto& b) -> bool {
+            std::ranges::sort(data.coalescables, [this] (const auto& a, const auto& b) -> bool {
```
```suggestion
            std::ranges::sort(data.coalescables, [this](const auto& a, const auto& b) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirAllocateRegistersByGreedy.cpp:
```diff
@@ -1181,15 +1180,14 @@ class GreedyAllocator {
 
-        std::sort(moves.begin(), moves.end(),
-            [](Move& a, Move& b) -> bool {
+        std::ranges::sort(moves, [](Move& a, Move& b) -> bool {
```
```suggestion
        std::ranges::sort(moves, [](auto& a, auto& b) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirAllocateStackByGraphColoring.cpp:
```diff
@@ -246,11 +246,9 @@ class GraphColoringStackAllocator final : public StackAllocatorBase {
-            [&] (CoalescableMove& a, CoalescableMove& b) -> bool {
-                return a.frequency > b.frequency;
-            });
+        std::ranges::sort(m_coalescableMoves, [&] (CoalescableMove& a, CoalescableMove& b) -> bool {
```
```suggestion
        std::ranges::sort(m_coalescableMoves, [&](auto& a, auto& b) {
```

weinig commented on Source/JavaScriptCore/b3/air/AirFixObviousSpills.cpp:
```diff
@@ -577,13 +577,13 @@ class FixObviousSpills {
         void sort()
         {
-            std::sort(regConst.begin(), regConst.end(), [] (const RegConst& a, const RegConst& b) {
+            std::ranges::sort(regConst, [] (const RegConst& a, const RegConst& b) {
```
Going to assume for the rest, you get the pattern.

```suggestion
            std::ranges::sort(regConst, [](auto& a, auto& b) {
```

weinig commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -1823,10 +1821,9 @@ static JSArray* availableNumberingSystems(JSGlobalObject* globalObject)
-            return WTF::codePointCompare(a, b) < 0;
-        });
+    std::ranges::sort(elements, [](const String& a, const String& b) {
+        return WTF::codePointCompare(a, b) < 0;
```
This can use WTF::codePointCompareLessThan to simplify it to:

```c++
        std::ranges::sort(elements, WTF::codePointCompareLessThan);
```

weinig commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -1773,10 +1772,9 @@ static JSArray* availableCurrencies(JSGlobalObject* globalObject)
-        [](const String& a, const String& b) {
-            return WTF::codePointCompare(a, b) < 0;
-        });
+    std::ranges::sort(elements, [](const String& a, const String& b) {
```
This can use WTF::codePointCompareLessThan to simplify it to:

```c++
        std::ranges::sort(elements, WTF::codePointCompareLessThan);
```

weinig commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -1881,10 +1878,9 @@ const Vector<String>& intlAvailableTimeZones()
-            [](const String& a, const String& b) {
-                return WTF::codePointCompare(a, b) < 0;
-            });
+        std::ranges::sort(temporary, [](const String& a, const String& b) {
```
This can use WTF::codePointCompareLessThan to simplify it to:

```c++
        std::ranges::sort(temporary, WTF::codePointCompareLessThan);
```

weinig commented on Source/JavaScriptCore/runtime/JSModuleNamespaceObject.cpp:
```diff
@@ -53,7 +53,7 @@ void JSModuleNamespaceObject::finishCreation(JSGlobalObject* globalObject, Abstr
     //
     // Sort the exported names by the code point order.
-    std::sort(resolutions.begin(), resolutions.end(), [](const auto& lhs, const auto& rhs) {
+    std::ranges::sort(resolutions, [](const auto& lhs, const auto& rhs) {
```
This can use WTF::codePointCompareLessThan and a projection to simplify it to:

```c++
std::ranges::sort(resolutions, WTF::codePointCompareLessThan, &std::pair::first);
```

weinig reviewed (approved):
I left a number of notes, but the only ones I strongly prefer you fixing are the style ones. Removing the space after the `[]`, and replacing unnecessary types in the lambda with auto.

For another PR, It would be good to standardize on the naming of the two parameters to these predicate lambdas, and then document that in the style guide. There are a good bunch of different ones, and that doesn't really serve anyone:

```
[](auto& a, auto& b) { ... }
[](auto& lhs, auto& rhs) { ... }
[](auto& left, auto& right) { ... }
```

We should probably find out what is the most prevalent and pick that.

iidmsa committed:
Use C++20 std::ranges::sort in JavaScriptCore
https://bugs.webkit.org/show_bug.cgi?id=302761
rdar://165017405

Reviewed by Sam Weinig.

* Source/JavaScriptCore/assembler/ARM64Assembler.h:
* Source/JavaScriptCore/assembler/ARMv7Assembler.h:
(JSC::ARMv7Assembler::jumpsToLink):
* Source/JavaScriptCore/b3/B3InferSwitches.cpp:
* Source/JavaScriptCore/b3/B3LowerMacros.cpp:
* Source/JavaScriptCore/b3/B3OptimizeAssociativeExpressionTrees.cpp:
(JSC::B3::OptimizeAssociativeExpressionTrees::optimizeRootedTree):
* Source/JavaScriptCore/b3/B3UseCounts.cpp:
(JSC::B3::UseCounts::UseCounts):
* Source/JavaScriptCore/b3/B3Validate.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackByLinearScan.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersByGreedy.cpp:
(JSC::B3::Air::Greedy::GreedyAllocator::finalizeGroups):
* Source/JavaScriptCore/b3/air/AirAllocateStackByGraphColoring.cpp:
* Source/JavaScriptCore/b3/air/AirFixObviousSpills.cpp:
* Source/JavaScriptCore/bytecode/CallLinkStatus.cpp:
(JSC::CallLinkStatus::computeFromCallLinkInfo):
* Source/JavaScriptCore/bytecode/PreciseJumpTargets.cpp:
(JSC::computePreciseJumpTargetsInternal):
* Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp:
(JSC::MetadataStatistics::reportMetadataStatistics):
* Source/JavaScriptCore/debugger/DebuggerParseData.cpp:
(JSC::DebuggerPausePositions::forEachBreakpointLocation):
(JSC::DebuggerPausePositions::sort):
* Source/JavaScriptCore/dfg/DFGCommonData.cpp:
(JSC::DFG::CommonData::finalizeCatchEntrypoints):
* Source/JavaScriptCore/dfg/DFGGraph.cpp:
(JSC::DFG::Graph::dumpBlockHeader):
* Source/JavaScriptCore/dfg/DFGJITCode.cpp:
(JSC::DFG::JITCode::finalizeOSREntrypoints):
* Source/JavaScriptCore/dfg/DFGLoopUnrollingPhase.cpp:
(JSC::DFG::LoopUnrollingPhase::populateCandidateLoops):
* Source/JavaScriptCore/dfg/DFGMinifiedGraph.cpp:
(JSC::DFG::MinifiedGraph::prepareAndShrink):
* Source/JavaScriptCore/dfg/DFGNode.h:
(JSC::DFG::nodeMapDump):
(JSC::DFG::nodeValuePairListDump):
* Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp:
* Source/JavaScriptCore/dfg/DFGScoreBoard.h:
(JSC::DFG::ScoreBoard::sortFree):
* Source/JavaScriptCore/dfg/DFGStoreBarrierClusteringPhase.cpp:
* Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp:
(JSC::FTL::DFG::LowerDFGToB3::compileCompareStrictEq):
* Source/JavaScriptCore/heap/HeapSnapshot.cpp:
(JSC::HeapSnapshot::finalize):
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
(JSC::HeapSnapshotBuilder::dumpToStream):
* Source/JavaScriptCore/heap/JITStubRoutineSet.cpp:
(JSC::JITStubRoutineSet::prepareForConservativeScan):
* Source/JavaScriptCore/heap/MarkedSpace.cpp:
* Source/JavaScriptCore/heap/MarkingConstraintSet.cpp:
(JSC::MarkingConstraintSet::executeConvergenceImpl):
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
(Inspector::JSInjectedScriptHost::queryHolders):
* Source/JavaScriptCore/jit/JITSizeStatistics.cpp:
(JSC::JITSizeStatistics::dump const):
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
(JSC::CompactTDZEnvironment::sortCompact):
* Source/JavaScriptCore/runtime/ArrayPrototype.cpp:
(JSC::sortBucketSort):
* Source/JavaScriptCore/runtime/ImportMap.cpp:
(JSC::ImportMap::ImportMap):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::flattenFields):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::availableCollations):
(JSC::availableCurrencies):
(JSC::availableNumberingSystems):
(JSC::intlAvailableTimeZones):
* Source/JavaScriptCore/runtime/JSModuleNamespaceObject.cpp:
(JSC::JSModuleNamespaceObject::finishCreation):
* Source/JavaScriptCore/runtime/JSONObject.cpp:
(JSC::logOutcomeImpl):
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::JSObject::getOwnIndexedPropertyNames):
* Source/JavaScriptCore/runtime/JSObjectInlines.h:
(JSC::JSObject::forEachOwnIndexedProperty):
* Source/JavaScriptCore/wasm/WasmMergedProfile.cpp:
(JSC::Wasm::MergedProfile::Candidates::finalize const):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::prepareForTailCallImpl):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::prepareForTailCallImpl):
* Source/JavaScriptCore/wasm/WasmOpcodeCounter.cpp:
(JSC::Wasm::WasmOpcodeCounter::dump):
* Source/JavaScriptCore/wasm/debugger/tests/TestUtilities.cpp:
(WasmDebugInfoTest::parseAndVerifyDebugInfoImpl):
* Source/JavaScriptCore/yarr/YarrJIT.cpp:
* Source/JavaScriptCore/yarr/YarrPattern.cpp:
(JSC::Yarr::CharacterClassConstructor::performSetOpWithMatches):
(JSC::Yarr::CharacterClassConstructor::sort):

Canonical link: https://commits.webkit.org/303399@main
